src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadmeControl.xaml.cs

```diff
@@ -0,0 +1,147 @@
+            _markdownPreview = new PreviewBuilder().Build();
+#pragma warning restore CS0618 // Type or member is obsolete
+            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        public DetailedPackageMetadata PackageMetadata
+        {
+            get
+            {
+                return (DetailedPackageMetadata)GetValue(PackageMetadataProperty);
+            }
+            set
+            {
+                UpdateControl(PackageMetadata, value);
+                SetValue(PackageMetadataProperty, value);
+            }
+        }
+
+        public void OnTabFilterChange(ItemFilter filter)
+        {
+            _renderLocalReadme = filter != ItemFilter.All;
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _loadCts.Cancel();
+                _loadCts.Dispose();
+                _markdownPreview?.Dispose();
+            }
+
+            _disposed = true;
+        }
+
+        private static void OnPropertyChanged(
+           DependencyObject dependencyObject,
+           DependencyPropertyChangedEventArgs e)
+        {
+            if (e.Property == PackageMetadataProperty && dependencyObject is PackageMetadataReadmeControl control)
+            {
+                control?.UpdateControl((DetailedPackageMetadata)e.OldValue, (DetailedPackageMetadata)e.NewValue);
+            }
+        }
+
+        private void UpdateControl(DetailedPackageMetadata oldValue, DetailedPackageMetadata newValue)
+        {
+            if (newValue is not null &&
+                (oldValue?.Id != newValue.Id ||
+                oldValue?.Version != newValue.Version))
+            {
+                var loadCts = new CancellationTokenSource();
+                var oldCts = Interlocked.Exchange(ref _loadCts, loadCts);
+                oldCts?.Cancel();
+                oldCts?.Dispose();
+
+                NuGetUIThreadHelper.JoinableTaskFactory
+                .RunAsync(async () =>
+                {
+                    await ReadmeViewModel.LoadReadmeAsync(newValue, _renderLocalReadme, _loadCts.Token);
+                    await UpdateMarkdownAsync();
+                })
+                .PostOnFailure(nameof(PackageMetadataReadmeControl));
+            }
+        }
+
+        private async Task UpdateMarkdownAsync()
+        {
+            var markdown = string.Empty;
+            if (ReadmeViewModel?.CanDetermineReadmeDefined == true)
+            {
+                markdown = string.IsNullOrWhiteSpace(ReadmeViewModel.ReadmeMarkdown)
+                    ? UI.Resources.Text_NoReadme
+                    : ReadmeViewModel.ReadmeMarkdown;
+            }
```

Comment:
If there's no Readme defined, should this be an `else` where Visibility is collapsed and ErrorLoadingReadme is false (not an error, just no readme available)?

Reply:
Yes, I updated it so we collapse the visibility of the tab if the markdown is empty